lib/gui.py

```diff
@@ -96,18 +96,101 @@ def on_submit():
     submit_button.grid(row=3, column=1)
 
 
+def make_port_scan(main_window):
+    font = ""
+    width = 800
+    height = 600
+    port_scan_window = tk.Toplevel(main_window, bg="#222", width=width, height=height)
+    port_scan_window.resizable(False, False)
+    port_scan_window.title("Port Scanner")
+
+    port_scan_frame = tk.Frame(port_scan_window)
+    port_scan_frame.pack(fill=tk.BOTH, expand=True)
+
+    host = tk.StringVar()
+    host.set("127.0.0.1")
+    port_range = tk.StringVar()
+    port_range.set("1-65535")
+    threads = tk.StringVar()
+    threads.set(multiprocessing.cpu_count())
+
+    scan_label = tk.Label(port_scan_frame, text="Select scanning options", bg="#222", fg="#FFF")
+    scan_label.pack(fill=tk.X, expand=True)
+
+    def validate_input(host_address, ports_range, n_threads):
+        host_address = host_address.replace("default: ", "")
+        ports_range = ports_range.replace("default: ", "")
+        n_threads = n_threads.replace("default: ", "")
+        import re
+
+        regex = "^((25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])$"
+        ports = ports_range.split("-")
+
+        lower_port = ports[0]
+        upper_port = ports[1]
+        if not re.search(regex, host_address):
+            messagebox.showerror("Error", "Invalid host adress")
+        elif not len(ports) == 2:
+            messagebox.showerror("Error", "Invalid port range")
+        elif not lower_port.isdigit() and int(lower_port) > 0:
```

Comment:
If condition chaining is a thing in python and is better than duplicating code like this. What if we want to change the error message? Yeah

Reply:
I kind of don't like having one if statement over 100 characters long but okay

Reply:
Just format it correctly. having one huge if statement is a bad thing. having one which is formatted over multiple lines is fine